Checksec.cpp

```diff
@@ -42,49 +69,22 @@ Checksec::Checksec(string filepath)
     // is too short to contain a reference to the IMAGE_LOAD_CONFIG_DIRECTORY.
     if (imageOptionalHeader.NumberOfRvaAndSizes < IMAGE_DIRECTORY_ENTRY_LOAD_CONFIG + 1) {
         cerr << "Warn: short image data directory vector (no load config?)" << "\n";
-        goto end;
-    }
-
-    // https://docs.microsoft.com/en-us/windows/desktop/api/winnt/ns-winnt-_image_data_directory
-    IMAGE_DATA_DIRECTORY dir = imageOptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_LOAD_CONFIG];
-
-    if (!dir.VirtualAddress || !dir.Size) {
-        cerr << "Warn: No IMAGE_LOAD_CONFIG_DIRECTORY in the PE" << "\n";
-        goto end;
+        return;
     }
 
-    // NOTE(ww): This always returns false, even when there definitely is an
-    // IMAGE_LOAD_CONFIG_DIRECTORY in the image. Microsoft never bothered
-    // to update the internal size check, and still compares it against 0x40
-    // (the Windows XP load config size).
-    // loadConfig_.Size = sizeof(dir.Size);
-    // if (!GetImageConfigInformation(&loadedImage, &loadConfig_)) {
-    //     cerr << "Warn: Couldn't retrieve IMAGE_LOAD_CONFIG_DIRECTORY: " << GetLastError() << "\n";
-    // }
-
-    IMAGE_SECTION_HEADER sectionHeader = {0};
-
-    // Find the section that contains the load config directory.
-    // This should always be .rdata, but who knows?
-    for (uint64_t i = 0; i < loadedImage.NumberOfSections; i++) {
-        if (loadedImage.Sections[i].VirtualAddress < dir.VirtualAddress
-            && loadedImage.Sections[i].VirtualAddress > sectionHeader.VirtualAddress)
-        {
-            sectionHeader = loadedImage.Sections[i];
-        }
+    ULONG loadConfigDirectoryEntrySize = 0u;
+    const PVOID loadConfigDirectoryEntryData = ImageDirectoryEntryToDataEx(
+            loadedImage.MappedAddress,
+            TRUE,
```

Comment:
Sorry, one last nit: This should be `FALSE`, since `MapAndLoad` calls `MapViewOfFile` internally. Looks like this causes sporadic issues as-is.

Reply:
Here we go